tests/test_exacttarget_discover.py

```diff
@@ -1,87 +1,115 @@
-import datetime
+from base import ExactTargetBase
 import tap_tester.connections as connections
 import tap_tester.menagerie as menagerie
 import tap_tester.runner as runner
 import os
-import unittest
-import pdb
-import json
-import requests
 
-
-class ExactTargetDiscover(unittest.TestCase):
+class ExactTargetDiscover(ExactTargetBase):
 
     def name(self):
         return "tap_tester_exacttarget_discover_v1"
 
-    def tap_name(self):
-        return "tap-exacttarget"
-
-    def setUp(self):
-        required_env = {
-            "TAP_EXACTTARGET_CLIENT_ID",
-            "TAP_EXACTTARGET_CLIENT_SECRET",
-            "TAP_EXACTTARGET_TENANT_SUBDOMAIN",
-            "TAP_EXACTTARGET_V2_CLIENT_ID",
-            "TAP_EXACTTARGET_V2_CLIENT_SECRET",
-            "TAP_EXACTTARGET_V2_TENANT_SUBDOMAIN",
-        }
-        missing_envs = [v for v in required_env if not os.getenv(v)]
-        if missing_envs:
-            raise Exception("set " + ", ".join(missing_envs))
-
-    def get_type(self):
-        return "platform.exacttarget"
-
     def get_credentials(self):
         return {
             'client_secret': os.getenv('TAP_EXACTTARGET_CLIENT_SECRET')
         }
 
-    def get_properties(self):
-        yesterday = datetime.datetime.now(datetime.timezone.utc) - datetime.timedelta(days=1)
-        return {
-            'start_date': yesterday.strftime("%Y-%m-%dT%H:%M:%SZ"),
-            'client_id': os.getenv('TAP_EXACTTARGET_CLIENT_ID')
-        }
+    def get_properties(self, *args, **kwargs):
+        props = super().get_properties(*args, **kwargs)
+        props.pop('tenant_subdomain')
+        return props
 
     def test_run(self):
+        """
+        Testing that discovery creates the appropriate catalog with valid metadata.
+            • Verify number of actual streams discovered match expected
+            • Verify the stream names discovered were what we expect
+            streams should only have lowercase alphas and underscores
```

Comment:
This assertion is missing

Reply:
Added this assertion but, skipped 'data_extension' streams, because they are the custom tables we create in marketing cloud UI and, the stream name will be the table name we set in the UI, as seen in our instance the table name is 'This is a test'.